### PR TITLE
[INLONG-5851][TubeMQ] Optimize while-sleep to ScheduledExecutorService in tubemq-manager

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Prepare a Pull Request
- Fixes https://github.com/apache/inlong/issues/5851

### Motivation
it is not recommended to execute a scheduled task by while-sleep. 

### Modifications
Replace while-sleep with ScheduledExecutorService.

### CODE REVIEW

File: inlong-tubemq/tubemq-manager/pom.xml [hunk 1]
```diff
@@ -176,6 +176,10 @@
             <groupId>jakarta.validation</groupId>
             <artifactId>jakarta.validation-api</artifactId>
         </dependency>
+      <dependency>
```
Comment (gosonzhang) [inline inlong-tubemq/tubemq-manager/pom.xml:179]: Need to add the LICENSE and NOTICE (if any) that the third party depends on in [1]

https://github.com/apache/inlong/blob/master/licenses/inlong-tubemq-manager/

File: inlong-tubemq/tubemq-manager/pom.xml [hunk 2]
```diff
@@ -176,6 +176,11 @@
...
+        <groupId>io.netty</groupId>
+        <artifactId>netty-common</artifactId>
+          <version>4.1.77.Final</version>
```
Comment (dockerzhang) [inline inlong-tubemq/tubemq-manager/pom.xml:182]: version is not needed here.

File: inlong-tubemq/tubemq-manager/pom.xml [hunk 3]
```diff
@@ -176,6 +176,10 @@
             <groupId>jakarta.validation</groupId>
             <artifactId>jakarta.validation-api</artifactId>
         </dependency>
+      <dependency>
+        <groupId>io.netty</groupId>
+        <artifactId>netty-common</artifactId>
```
Comment (healchow) [inline inlong-tubemq/tubemq-manager/pom.xml:181]: Excuse me, where did you use the netty-common package?

File: licenses/inlong-tubemq-manager/LICENSE [hunk 4]
```diff
@@ -452,6 +452,7 @@ The text of each license is also included at licenses/LICENSE-[project].txt.
   org.springframework:spring-web:5.3.20 - Spring Web (https://github.com/spring-projects/spring-framework), (Apache License, Version 2.0)
   org.springframework:spring-webmvc:5.3.18 - Spring Web MVC (https://github.com/spring-projects/spring-framework), (Apache License, Version 2.0)
   org.apache.tomcat.embed:tomcat-embed-core:9.0.60 - tomcat-embed-core (https://tomcat.apache.org/), (Apache License, Version 2.0)
+  io.netty:netty-common:4.1.72.Final - Netty/Common (https://github.com/netty/netty/tree/netty-4.1.72.Final), (Apache License, Version 2.0)
```
Comment (healchow) [inline licenses/inlong-tubemq-manager/LICENSE:455]: If the netty-common jar is needed, it should be sorted alphabetically by artifactId, here it's `netty-common`.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-tubemq/tubemq-manager/pom.xml (hunk 3 above)
Comment (liangyepianzhou) [reply inlong-tubemq/tubemq-manager/pom.xml:181]: The DefaultThreadFactory in TopicBackendWorker.java

File: inlong-tubemq/tubemq-manager/src/main/java/org/apache/inlong/tubemq/manager/service/TopicBackendWorker.java [hunk 5]
```diff
@@ -136,5 +140,6 @@ public void run() {
     public void destroy() throws Exception {
         runFlag.set(false);
         nodeService.close();
+        this.workerExecutor.shutdown();
```
Comment (healchow) [inline inlong-tubemq/tubemq-manager/src/main/java/org/apache/inlong/tubemq/manager/service/TopicBackendWorker.java:143]: Maybe use `shutdownNow()`?

File: inlong-tubemq/tubemq-manager/src/main/java/org/apache/inlong/tubemq/manager/service/TopicBackendWorker.java [hunk 6]
```diff
@@ -67,9 +71,10 @@ public class TopicBackendWorker implements DisposableBean, Runnable  {
 
     TopicBackendWorker() {
         Thread thread = new Thread(this);
-        // daemon thread
-        thread.setDaemon(true);
-        thread.start();
+        this.workerExecutor = Executors
+                .newSingleThreadScheduledExecutor(
```
Comment (healchow) [inline inlong-tubemq/tubemq-manager/src/main/java/org/apache/inlong/tubemq/manager/service/TopicBackendWorker.java:75]: Introducing a package just to use a ThreadFactory is not a good solution.

It is recommended to use the `com.google.common.util.concurrent.ThreadFactoryBuilder` class from the guava package to create a ThreadFactory.

**Review by healchow — CHANGES_REQUESTED** [review]

File: licenses/inlong-tubemq-manager/NOTICE [hunk 7]
```diff
@@ -494,6 +494,10 @@ Jakarta Annotations API NOTICE
 Jakarta Persistence API NOTICE
 ------------------------------------------------------
 javax.transaction API NOTICE
+------------------------------------------------------
+Netty NOTICE
```
Comment (healchow) [inline licenses/inlong-tubemq-manager/NOTICE:498]: Please remove those notices.

**Review by healchow — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**